### PR TITLE
Do not crash when configuring s390 devices

### PR DESCRIPTION
**Pull request by mchf**
bnc#983201

### CODE REVIEW

File: src/modules/LanItems.rb [hunk 1]
```diff
@@ -2203,17 +2203,24 @@ def enableCurrentEditButton
 
     # Creates eth emulation for s390 devices
     #
-    # @param [Hash] an s390 device description as obtained from AY profile
-    def createS390Device(rule)
+    # @param [Hash] an s390 device description (e.g. as obtained from AY profile).
+    # if not present then initialization of S390 specific LanItems' variables is
+    # expected to be done out of this function.
+    def createS390Device(dev_attrs = [])
```
Comment (jreidinger) [inline src/modules/LanItems.rb]: param is hash and default value is array? it smell for me

Comment (mchf) [inline src/modules/LanItems.rb]: you're right. That happens when working in hurry. Thanks for catching that.

**Comment by jreidinger** [comment]
LGTM

File: src/modules/LanItems.rb [hunk 2]
```diff
@@ -2203,17 +2203,24 @@ def enableCurrentEditButton
 
     # Creates eth emulation for s390 devices
     #
-    # @param [Hash] an s390 device description as obtained from AY profile
-    def createS390Device(rule)
+    # @param [Hash] an s390 device description (e.g. as obtained from AY profile).
+    # if not present then initialization of S390 specific LanItems' variables is
+    # expected to be done out of this function.
+    def createS390Device(dev_attrs = {})
       Builtins.y2milestone("creating device s390 network device, #{rule}")
```
Comment (mvidner) [inline src/modules/LanItems.rb]: NameError

File: src/modules/LanItems.rb [hunk 3]
```diff
@@ -2203,17 +2203,24 @@ def enableCurrentEditButton
 
     # Creates eth emulation for s390 devices
     #
-    # @param [Hash] an s390 device description as obtained from AY profile
-    def createS390Device(rule)
+    # @param [Hash] an s390 device description (e.g. as obtained from AY profile).
+    # if not present then initialization of S390 specific LanItems' variables is
```
Comment (mvidner) [inline src/modules/LanItems.rb]: "Not present" is slightly ambiguous, I wouldn't know whether that means `empty?` or `nil?` -> "If empty then..."

File: src/modules/LanItems.rb (hunk 2 above)
Comment (mvidner) [inline src/modules/LanItems.rb]: BTW I can make ruby-lint find this bug, as well as the one this PR is fixing!
But they are hidden among 688 false positives.

File: src/modules/LanItems.rb [hunk 4]
```diff
@@ -2203,17 +2203,26 @@ def enableCurrentEditButton
...
-      @type = rule["type"] || ""
-      @qeth_chanids = rule["chanids"] || ""
-      @qeth_layer2 = rule.fetch("layer2", false)
-      @qeth_portname = rule["portname"] || ""
-      @chan_mode = rule["protocol"] || ""
-      @iucv_user = rule["router"] || ""
+    # @param [Hash] an s390 device description (e.g. as obtained from AY profile).
+    # If it contains s390 device attributes definition, then these definitions takes
+    # precendence over values assigned to corresponding LanItems' global variables
+    # before the method invocation. Hash keys are symbols named after LanItems'
```
Comment (mvidner) [inline src/modules/LanItems.rb]: strings, not symbols

**Comment by mvidner** [comment]
Apart from the comment mistake LGTM.

**Comment by mvidner** [comment]
LGTM.

**Comment by mchf** [comment]
Thank you for the review.

**Pull request merged**